hangman.py

```diff
@@ -0,0 +1,270 @@
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    my_word = my_word.replace(" ", "")
+    # This loop compares my_word, which consists of "_ " and letters which have already been guessed, with words
+    # from file "words.txt". If these two words have the same length and letters from my_word have the same
+    # position in the compared word, loop returns True. Otherwise, it returns False.
+    if len(my_word) == len(other_word):
+        i = 0
+        while i < len(my_word):
+            if my_word[i] == "_":
+                i += 1
+            else:
+                if my_word[i] != other_word[i]:
+                    return False
+                else:
+                    i += 1
+        return True
+    return False
+
+
+def show_possible_matches(my_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    returns: nothing, but should print out every word in wordlist that matches my_word
+             Keep in mind that in hangman when a letter is guessed, all the positions
+             at which that letter occurs in the secret word are revealed.
+             Therefore, the hidden letter(_ ) cannot be one of the letters in the word
+             that has already been revealed.
+
+    '''
+    matches = [word for word in wordlist if match_with_gaps(my_word, word)]
+    if not matches:
+        print("No matches found.\n")
+    else:
+        print("Possible word matches are: " + ", ".join(matches), end="." + "\n")
+    return
+
+
+def hangman_with_hints(secret_word):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Make sure to check that the user guesses a letter
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    * If the guess is the symbol *, print out all words in wordlist that
+      matches the current guessed word.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    print("Welcom to the game Hangman!\n" + "I am thinking of a word that is", len(secret_word), "letters long.")
```

Comment:
So you print this message only for game with hints? What user should see in game without hints?